iree/compiler/Dialect/Flow/Transforms/SplitReduction.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//===--------------- SplitReduction.cpp ----------------------------===//
+//
+// Split reduction dimension to increase parallelism of a linalg operation.
+//
+//===----------------------------------------------------------------------===//
+
+#include "iree/compiler/Dialect/Flow/Transforms/PassDetail.h"
+#include "iree/compiler/Dialect/Flow/Transforms/Passes.h"
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace IREE {
+namespace Flow {
+
+// TODO(thomasraoux): Move to attributes.
+static llvm::cl::opt<int64_t> splitReductionRatio(
```

Comment:
could this be a pass option instead of a command-line flag?

Reply:
I need to access it from iree-translate for now so it cannot be a pass option